applications/FreeSurfaceApplication/custom_utilities/edgebased_levelset_substep.h

```diff
@@ -2082,1898 +2066,1900 @@ class EdgeBasedLevelSetSubstep
+            {
+                if (mis_slip[i_node] == true)
+                    tempmSlipBoundaryList.push_back(i_node);
+                mis_slip[i_node] = false;
             }
+            mSlipBoundaryList.resize(tempmSlipBoundaryList.size(), false);
+#pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(tempmSlipBoundaryList.size()); i++)
+                mSlipBoundaryList[i] = tempmSlipBoundaryList[i];
 
+            // check that all of the normals are not zero
+            for (int i = 0; i < static_cast<int>(mSlipBoundaryList.size()); i++)
+            {
+                unsigned int i_node = mSlipBoundaryList[i];
+                double tmp = norm_2(mSlipNormal[i_node]);
+                if (tmp < 1e-20)
+                    KRATOS_THROW_ERROR(std::logic_error, "found a slip node with zero normal on node with id", i_node + 1)
+            }
 
-        //loop over all faces to fill inlet outlet
-        for (ModelPart::ConditionsContainerType::iterator cond_it = rConditions.begin(); cond_it != rConditions.end(); cond_it++)
-        {
-            //get geometry data of the face
-            Geometry<Node < 3 > >& face_geometry = cond_it->GetGeometry();
-            //reference for area normal of the face
-            array_1d<double, 3 > & face_normal = cond_it->GetValue (NORMAL);
-	    bool is_inlet_or_outlet = false;
-            if (cond_it->GetValue (IS_STRUCTURE) == 0) is_inlet_or_outlet = true;
-	    else
-	    {
-	      for (unsigned int if_node = 0; if_node < TDim; if_node++)
-                    if (face_geometry[if_node].IsFixed (VELOCITY_X)  )
-		    is_inlet_or_outlet = true;
-	    }
-            //slip condition
-            if (is_inlet_or_outlet) //the opposite of the loop before
-                for (unsigned int if_node = 0; if_node < TDim; if_node++)
-                {
-                    unsigned int i_node = static_cast<unsigned int> (face_geometry[if_node].FastGetSolutionStepValue (AUX_INDEX) );
-                    array_1d<double, TDim>& inout_normal = mInOutNormal[i_node];
-                    mis_slip[i_node] = true; //reutilize it!
+            // loop over all faces to fill inlet outlet
+            for (ModelPart::ConditionsContainerType::iterator cond_it = rConditions.begin(); cond_it != rConditions.end(); cond_it++)
+            {
+                // get geometry data of the face
+                Geometry<Node<3>> &face_geometry = cond_it->GetGeometry();
+                // reference for area normal of the face
+                array_1d<double, 3> &face_normal = cond_it->GetValue(NORMAL);
+                bool is_inlet_or_outlet = false;
+                if (cond_it->IsNot(SLIP))
+                    is_inlet_or_outlet = true;
+                else
+                {
+                    for (unsigned int if_node = 0; if_node < TDim; if_node++)
+                        if (face_geometry[if_node].Is(INLET) || face_geometry[if_node].Is(OUTLET))
```

Comment:
Same observation about the outlet check,

Reply:
Done